src/siemens-s7_fmt_plug.c

```diff
@@ -93,6 +93,10 @@ static int valid(char *ciphertext, struct fmt_main *self)
 		goto bail;
 	if (strlen(p) != 40)
 		goto bail;
+	if ((p = strtok(NULL, "$")) == NULL)	/* Fix bug: #1090 */
+		goto bail;
+	if (strlen(p) < 40)
+		goto bail;
```

Comment:
@magnumripper  I am not sure whether using if (strlen(p) != 40). But I can sure strlen(p) must longer than or equal 40. I will send a mail to the author to make sure.